### PR TITLE
Ross Nichols (Fatima) submission for callbacks and iterators deliverable

### PR DESCRIPTION
**Pull request by nichro02**
I spent an unhealthy amount of time working on this deliverable, map #2 in particular. I felt like I spent way too much time spinning my wheels to trying get a conditional to work, but ultimately found a way to get the code to run so that what logged to console looked at least somewhat correct. I felt like I understood what the map function is supposed to do, but didn't feel too confident in being able to ultimately implement the concept. Filter #2 also gave me a bit of trouble, but felt a lot better about that one after looking at it again.

### CODE REVIEW

File: map2.js [hunk 1]
```diff
@@ -31,3 +31,18 @@ var phoneBook = [
...
+  //converts key to string
+  const num = Object.values(item).toString()
+  //converts value to string
+  const firstNum = num.charAt()
+  //grab character at index 0
+  const concNum = firstNum+'-'+num
+  //create a new phone number that appends prefix captured in firstNum
+  return {[strName]: concNum}
+  //save object that has key of person's name and value of updated phone number to the new array
+})
```
Comment (ghost) [inline map2.js:45]: great work with this! your comments really help elucidate what you were thinking, and your reasoning is 💯 . 

I ran the code here and it's interesting to me that you put `strName` in square brackets in the return statement. I expected it to turn the key into an array, but it didn't; learned something new today 😄 . That said, the square brackets in line 43 aren't necessary. Out of curiosity, what was your reasoning for putting them in?

**Comment by ghost** [comment]
I want to address your comment about spending an "unhealthy amount of time" on a problem to reassure you -- it's totally fine if you're spending a long time on a problem! That's how you learn, by puzzling over things, and I can tell through your comments in your code that you're really taking your time to understand how things are working. That is **the best thing** you could be doing. 

It also took me several hours to figure out homework when I was a student, and not just during unit 1. Your dedication and perseverance will pay dividends, because you're spending time learning how to learn how to code. That is what will help you learn material far beyond the scope of the course. So keep going, you're doing everything exactly right 🥇

**Comment by ghost** [comment]
oh, yeah, and I'm marking this complete 😄

File: map2.js (hunk 1 above)
Comment (nichro02) [reply map2.js:45]: I can't remember why I added square brackets - maybe my mind was on arrays? I'll go in and update that. Thanks for the feedback!

**Pull request closed without merging**